srtcore/api.cpp

```diff
@@ -233,15 +237,24 @@ string srt::CUDTUnited::CONID(SRTSOCKET sock)
     return os.str();
 }
 
-int srt::CUDTUnited::startup()
+void srt::CUDTUnited::implicitStartup()
 {
     ScopedLock gcinit(m_InitLock);
     if (m_bGCStatus)
-        return 1;
+        return;
+    m_bImplicitStartup = true;
+    startup();
```

Comment:
Double locking on `m_InitLock` here.

Reply:
The "lock primitive used", if you look at that from the C++11 perspective, is `std::mutex`, and no, it is not recursive. Recursive mutex in C++11 is `std::recursive_mutex`, but we don't use it and we don't have its equivalent in the POSIX wrapper version.

Reply:
ah ok, I'll fix.